### ISSUE TITLE
Get vulnerability score/severity with pip-audit

### ISSUE THREAD

**Issue by matteoannotell** [trailofbits/pip-audit#654]
**Is your feature request related to a problem? Please describe.**

We would like to be able to see the severity of vulnerabilities detected, and ideally filter them out based on being above/below a given threshold. 

**Describe the solution you'd like**

Adding score/severity the information to the output would be a first step for enabling the user to ideally pipe the output to a json and have some case/if downstream based on that.
Ideally one could go even further and perhaps pass a `--threshold` flag for muting vulnerabilities below a given score

**Describe alternatives you've considered**

I drafted a PR for addressing the first part of the problem, providing score/severity, https://github.com/pypa/pip-audit/pull/653.
An alternative would be using pip-audit as it is, and calling the nvd api using pip-audit output as input. The main risk here is creating a dependency on a third-party api

**Additional context**
```
INFO:pip_audit._cli:This product uses the NVD API but is not endorsed or certified by the NVD.
Found 2 known vulnerabilities in 2 packages
Name       Version ID               Fix Versions Severity Score
---------- ------- ---------------- ------------ -------- -----
py         1.11.0  PYSEC-2022-42969              HIGH     7.5
setuptools 65.5.0  PYSEC-2022-43012 65.5.1       MEDIUM   5.9
```

**Comment by matteoannotell** [comment]
Good feedback William, thanks! 
I understand the concerns: I will then close the PR, you can disregard it, but thanks a lot for your offer to review it 😄 

At the best of your knowledge, do you know if there is any plans, or if there has been any discussions in either PyPI or OSV for adding some sort of vulnerability info?

Also, not fully familiar with how PYSEC/CVE, and GHSA relate to each other. Is it possible that a PYSEC ID has neither a GHSA nor a CVE counterpart?

**Comment by di** [comment]
I think the upstream issue in OSV is https://github.com/google/osv.dev/issues/783, PyPI would be able to add CVSS scores once OSV supported it.

**Comment by woodruffw** [comment]
>  Is it possible that a PYSEC ID has neither a GHSA nor a CVE counterpart?

It's currently unlikely, but it's technically possible: PYSEC IDs come from a [separate database](https://github.com/pypa/advisory-database) which is back-populated with relevant CVEs, but it's possible that future vulnerabilities are "unique" to PYSEC or may appear in PYSEC before appearing in other feeds (roughly analogously to RUSTSEC).

**Comment by matteoannotell** [comment]
I understand. Thank you both for the info!

**Comment by dpinol** [comment]
https://github.com/google/osv.dev/issues/783 looks like it's already implemented

**Comment by guerda** [comment]
The OSV API delivers CVE scores already, so if a CVE is coming from OSV, pip audit could read it from the body:

<img width="369" height="375" alt="Image" src="https://github.com/user-attachments/assets/23181da6-b68b-477c-a391-80d872e1d9eb" />
<img width="914" height="196" alt="Image" src="https://github.com/user-attachments/assets/861df027-8416-4736-8431-d629b89def57" />

The PyPI API does not deliver scores yet:
Example call: https://pypi.org/pypi/aiosmtplib/3.0.0/json

I have not found any issue in the pypi organization which request inclusion of CVE scores in their api.

**Comment by woodruffw** [comment]
Yes, we'd need this to be exposed in PyPI first. Our rule of thumb is that the information exposed by PyPI is our baseline/common subset for all vulnerability service backends supported by pip-audit.

**Comment by guerda** [comment]
I opened a feature request for pypi.org, let's see if it gets implemented: https://github.com/pypi/warehouse/issues/20313
